### ISSUE TITLE
CI checks are hostile to forks

### ISSUE THREAD

**Issue by jsoref** [vectordotdev/vector#15971]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L90
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L120
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L148
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L175
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L257
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L268
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L280
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L292
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L304
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/test.yml#L315

https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/regression.yml#L88
https://github.com/vectordotdev/vector/blob/34af7e601cc043635efd1081b5a2823c8f225092/.github/workflows/regression.yml#L125

### Configuration

```text
A fork of this repository with a PR, e.g. https://github.com/check-spelling/vector/pull/1
```

### Version

94fcd43bbe562e290ba277b1e78d324cd6000f31

### Debug Output

```text
https://github.com/check-spelling/vector/actions/runs/3928407787/jobs/6716008492

> Unit - x86_64-unknown-linux-gnu
Started 16h 56m 27s ago
Requested labels: linux, test-runner
Job defined at: check-spelling/vector/.github/workflows/test.yml@refs/pull/1/merge
Waiting for a runner to pick up this job...

> Build baseline Vector container
Started 16h 57m 20s ago
Requested labels: linux, soak-builder
Job defined at: check-spelling/vector/.github/workflows/regression.yml@refs/pull/1/merge
Waiting for a runner to pick up this job...
```

### Additional Context

It's possible to write CI jobs that will either not run in forks or will use github's provided runners as a fallback.

**Comment by jszwedko** [comment]
Hi @jsoref !

I appreciate the report. What would be the proposed change though? We need to run the tests on more powerful GHA runners and I'm not aware of a way to provide a fallback for `runs-on`.

**Comment by jsoref** [comment]
You can do something like this:
https://github.com/commercialhaskell/stack/blob/783f195401f4eaa4ae69deb11751860b98643080/.github/workflows/integration-tests.yml#L145-L172

The precise details will obviously vary based on what you need to do.

**Comment by jsoref** [comment]
If your tests will fail on the GitHub runners, then you can use `if` to let the jobs not run on forks.

**Comment by jszwedko** [comment]
> You can do something like this: https://github.com/commercialhaskell/stack/blob/783f195401f4eaa4ae69deb11751860b98643080/.github/workflows/integration-tests.yml#L145-L172
> 
> The precise details will obviously vary based on what you need to do.

Nifty, I see. I didn't realize `runs-on` could use outputs too. Something like that seems like it would work.

**Comment by jsoref** [comment]
Btw, if you can identify the documentation this project looked at in order to set up this workflow, I'd be happy to file a ticket to update it to suggest doing something like this -- it's a lot easier to fix one or two docs than lots of repositories :).

**Comment by jszwedko** [comment]
> Btw, if you can identify the documentation this project looked at in order to set up this workflow, I'd be happy to file a ticket to update it to suggest doing something like this -- it's a lot easier to fix one or two docs than lots of repositories :).

Very true :) However I think the workflow definitions are mostly home grown from following the GitHub Actions documentation. Clearly we were not considering if they would be runnable on forks though. I think this is the first time we've had someone ask about running CI on a fork 😅

**Comment by spencergilbert** [comment]
Yeah, I don't think any of us claim to be a GH Actions expert - definitely been a making it work as we go situation.

Super happy to take any feedback/suggestions/etc

**Comment by thomasqueirozb** [comment]
Most of these are using `ubuntu-24.04` now. Only publish is using custom runners (no fork should want to use this job) and some runners are using `ubuntu-24.04-8core`/`windows-2025-8core`, which are just the regular runners using 8 cores. The "fix" for a fork to run most tests is to run `sed -i 's/-8core//g .github/workflows/*.yml` or actually create these runners which will take an admin maybe 5 minutes.

I think this can be marked as complete now
